curvefs/src/client/s3/client_s3_adaptor.h

```diff
@@ -65,10 +65,21 @@ class S3ClientAdaptor {
      * @brief Initailize s3 client
      * @param[in] options the options for s3 client
      */
-    virtual CURVEFS_ERROR Init(const S3ClientAdaptorOption& option,
-                               S3Client* client,
-                               std::shared_ptr<InodeCacheManager> inodeManager,
-                               std::shared_ptr<MdsClient> mdsClient) = 0;
+    virtual CURVEFS_ERROR
+    Init(const S3ClientAdaptorOption &option, std::shared_ptr<S3Client> client,
```

Comment:
Why S3ClientAdaptorImpl need so many Init function?

Reply:
Distinguish whether diskcache is required by whether diskCacheManagerImpl is null or not